main.tf

```diff
@@ -339,6 +339,25 @@ resource "aws_cloudfront_distribution" "default" {
 
   aliases = var.acm_certificate_arn != "" ? var.aliases : []
 
+  dynamic "origin_group" {
+    for_each = var.origin_groups
+    content {
+      origin_id = "${module.this.id}-group[${origin_group.key}]"
+
+      failover_criteria {
+        status_codes = origin_group.value.failover_criteria
+      }
+
+      member {
+        origin_id = try(length(origin_group.value.primary_origin_id), 0) > 0 ? origin_group.value.primary_origin_id : module.this.id
```

Comment:
I see this in the variable explanation.

> If `primary_origin_id` is set to `null` or `""`, then the origin id of the origin created by this module will be used in its place.

The variable `origin_groups` requires the `primary_origin_id` and `failover_origin_id`. The above line says that if the `primary_origin_id` is set, then it uses the `primary_origin_id` otherwise the context's id. Could you explain why this is so?

Also, how come the `failover_origin_id` doesn't get the same benefit ?